### ISSUE TITLE
Pyright crashes with internal error

### ISSUE THREAD

**Issue by BrandonNav** [microsoft/pyright#7828]
**Describe the bug**
I run pyright on my codebase and it crashes with the following stacktrace

```
An internal error occurred while type checking file "/Users/brandon/Work/code/server/core/paginator.py": Error: Debug Failure. False expression.
    at evaluateTypeParameterList (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:17135:9)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16280:34)
    at a.getTypeOfClass [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.visitClass (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:309:49)
    at q.visit (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:388:29)
    at q.visitNode (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:283:23)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20476:28)
    at a.suppressDiagnostics [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:282:29)
    at q._walkStatementsAndReportUnreachable (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2822:18)
    at q.visitSuite (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:289:14)
    at q.visit (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:280:19)
    at forEach (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at q.walkMultiple (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:280:19)
    at q._walkStatementsAndReportUnreachable (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2822:18)
    at q.check (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:261:14)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:884:29)
    at a.timeOperation (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:44:28)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:875:45)
    at t.LogTracker.log (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/logTracker.ts:49:20)
    at t.SourceFile.check (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:873:33)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1928:44)
    at t.LogTracker.log (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/logTracker.ts:49:20)
    at F._checkTypes (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1884:33)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:648:30)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:633:20)
    at a.runWithCancellationToken [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:44:28)
    at F._runEvaluatorWithCancellationToken (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1037:41)
    at F.analyze (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:612:21)
    at analyzeProgram (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/analysis.ts:52:33)
    at t.BackgroundAnalysisProgram.startAnalysis (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/backgroundAnalysisProgram.ts:157:16)
    at Timeout._onTimeout (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/service.ts:1719:67)
    at listOnTimeout (node:internal/timers:569:17)
    at processTimers (node:internal/timers:512:7)
Error performing analysis: Error: Debug Failure. False expression.
    at evaluateTypeParameterList (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:17135:9)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16280:34)
    at a.getTypeOfClass [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.visitClass (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:309:49)
    at q.visit (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:388:29)
    at q.visitNode (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:283:23)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20476:28)
    at a.suppressDiagnostics [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:282:29)
    at q._walkStatementsAndReportUnreachable (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2822:18)
    at q.visitSuite (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:289:14)
    at q.visit (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:280:19)
    at forEach (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at q.walkMultiple (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at q.walk (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:280:19)
    at q._walkStatementsAndReportUnreachable (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2822:18)
    at q.check (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:261:14)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:884:29)
    at a.timeOperation (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:44:28)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:875:45)
    at t.LogTracker.log (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/logTracker.ts:49:20)
    at t.SourceFile.check (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/sourceFile.ts:873:33)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1928:44)
    at t.LogTracker.log (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/logTracker.ts:49:20)
    at F._checkTypes (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1884:33)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:648:30)
    at callback (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:633:20)
    at a.runWithCancellationToken [as timeOperation] (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:44:28)
    at F._runEvaluatorWithCancellationToken (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:1037:41)
    at F.analyze (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/program.ts:612:21)
    at analyzeProgram (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/analysis.ts:52:33)
    at t.BackgroundAnalysisProgram.startAnalysis (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/backgroundAnalysisProgram.ts:157:16)
    at Timeout._onTimeout (/Users/brandon/.npm/_npx/fbb43b1786f81b3f/node_modules/pyright/dist/pyright-internal/src/analyzer/service.ts:1719:67)
    at listOnTimeout (node:internal/timers:569:17)
    at processTimers (node:internal/timers:512:7)
```

**Code or Screenshots**
I have reduced my code to the following:

```python
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from django.core.paginator import _SupportsPagination
    from django.db.models import Model
else:

    class _SupportsPagination[T]:
        pass


class SupportsModelPagination(_SupportsPagination[Model]):
    pass
```

You will need `django-stubs==4.2.7` installed

**VS Code extension or command-line**
This is crashing in both the pylance extension and the command-line.

```
npx pyright@latest server/core/paginator.py
-> crashes

npx pyright@1.1.360 server/core/paginator.py
-> crashes

npx pyright@1.1.359 server/core/paginator.py
-> 0 errors, 0 warnings, 0 informations
```

Node version: v18.16.0

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.362, which I just published.
